tests/unit/ClientTest.php

```diff
@@ -226,4 +226,39 @@ public function testSet()
         $message = $this->connection->getLastMessage();
         $this->assertEquals('test.barfoo:666|s', $message);
     }
+
+    /**
+     * @dataProvider tagsProvider
+     */
+    public function testTags($tags, $expectedTagMessage)
+    {
+        $this->client->set("foobaz", 999, $tags);
+
+        $message = $this->connection->getLastMessage();
+        $this->assertEquals('test.foobaz:999|s|#' . $expectedTagMessage, $message);
+    }
+
+    /**
+     * Data provider for testTags
+     * @return array
+     */
+    public function tagsProvider()
+    {
+        return [
+          // single string tag
+          ['one', 'one'],
+
+          // basic arrays
+          [['one'], 'one'],
+          [['one', 'two'], 'one,two'],
```

Comment:
What's expected when values are duplicated? Might not be a valid use case. Worth testing? e.g. `[['one', 'one'], 'one,one'],`